Habitica/src/main/java/com/habitrpg/android/habitica/ui/fragments/preferences/PreferencesFragment.java

```diff
@@ -197,6 +204,36 @@ public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, Strin
             } else {
                 pushNotificationManager.removePushDeviceUsingStoredToken();
             }
+        } else if (key.equals("language")) {
+            LanguageHelper languageHelper = new LanguageHelper(sharedPreferences.getString(key,"en"));
+            Locale.setDefault(languageHelper.getLocale());
+            Configuration configuration = new Configuration();
+
+            if (android.os.Build.VERSION.SDK_INT <= Build.VERSION_CODES.JELLY_BEAN){
+                configuration.locale = languageHelper.getLocale();
+            } else {
+                configuration.setLocale(languageHelper.getLocale());
+            }
+
+            getActivity().getResources().updateConfiguration(configuration,
+                    getActivity().getResources().getDisplayMetrics());
```

Comment:
What content are you referring to? In the latest commit, the app reloads immediately after the language change so everything is displayed in the new language right after the user changes it. I don't see any content that remains in the old language after the user has chosen a new one.